linalg/dtensor.hpp

```diff
@@ -111,6 +111,9 @@ class DeviceTensor
       data = t.data;
    }
 
+   /// Copy assignment not supported
+   MFEM_HOST_DEVICE DeviceTensor& operator=(DeviceTensor&) = delete;
```

Comment:
I ran into a problem when this got deleted. I get the following error from our RAJA/CAMP kernels: 

```
camp-3fb4621/include/camp/tuple.hpp:205:99: error: the parameter for this explicitly-defaulted copy assignment operator is const, but a member or base requires it to be non-const
renamed_intern_namespace::template tuple_helper< camp::idx_seq< Indices...> , list< Types...> >  &operator=(const renamed_intern_namespace::template tuple_helper< camp::idx_seq< Indices...> , list< Types...> >  & rhs) = default;
```

If I remove line 115 my code builds just fine. I'm not sure what would be appropriate here. ping @trws , @v-dobrev

Reply:
For this class we can actually just ` = default` both the copy constructor and the copy assignment. Currently, the copy constructor does what the `= default` version will do.

Reply:
Can you make this change in this branch @v-dobrev ?

Reply:
The error message is actually about the input parameter (r.h.s.) not being `const` -- we should fix that too.

Reply:
First try changing line 115 to accept a `const DeviceTensor&` and see if it makes a difference.  The signature of a copy assignment deletion should always have the const in it, I'm not sure what this version will do.

Aside from that, if copy assignment isn't supported, and someone tries to copy assign a tuple containing one of these, it is supposed to error out.  We'd need to figure out what contracts to enforce here to figure out how exactly to fix it if it isn't just a type mismatch mistake.

Reply:
Updated.

Reply:
Using `   MFEM_HOST_DEVICE DeviceTensor& operator=(const DeviceTensor& ) = default;` worked for me. I can try switching default to be delete and trying that as well.

Reply:
It looks like `nvcc` issues warnings with `__host__ __device__` and `= default` -- I'll try removing `__host__ __device__` to see that works without warnings.

Reply:
It should stop the warnings, but we've seen a few times that it will spuriously frail to generate the default the right way.  If that happens, it may have to be explicit for a while until updating to a sufficiently new nvcc.

Reply:
Seems to be okay with CUDA 10.1.243. I'll just drop the `MFEM_HOST_DEVICE`. Hopefully HIP will be okay with that too.

Reply:
Pushed.

Reply:
#TeamWork!